### PR TITLE
Add v2 API to FileParsers

### LINKED ISSUE #7074: [v2 API] FileParsers
Add the new v2 API for the `FileParsers` library

### PR DESCRIPTION
**Pull request by greglandrum**
This applies the same ideas used for the SMILES and SMARTS parsers

Fixes #7074

### CODE REVIEW

File: Code/GraphMol/FileParsers/Mol2FileParser.cpp [hunk 1]
```diff
@@ -902,48 +901,32 @@ RWMol *Mol2DataStreamToMol(std::istream *inStream, bool sanitize, bool removeHs,
...
   } else {
-    inStream->seekg(chargeStart, std::ios::beg);
-    try {
-      readFormalChargesFromAttr(inStream, res);
-    } catch (const FileParseException &e) {
-      delete res;
-      throw e;
-    }
+    inStream.seekg(chargeStart, std::ios::beg);
+    readFormalChargesFromAttr(&inStream, res.get());
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/Mol2FileParser.cpp:929]: Should we make a v2 readFormalCharges that takes a reference to a mol not a pointer?

```
readformalChargesFromAttr(&inStream, *res);
```

File: Code/GraphMol/FileParsers/MolFileParser.cpp [hunk 2]
```diff
@@ -3256,30 +3255,24 @@ void finishMolProcessing(RWMol *res, bool chiralityPossible, bool sanitize,
...
+      // bonds (e.g. imines). But before we run stereo detection,
+      // we need to run mol cleanup so don't have trouble with
+      // e.g. nitro groups. Sadly, this a;; means we will find
+      // run both cleanup and ring finding twice (a fast find
+      // rings in bond stereo detection, and another in
+      // sanitization's SSSR symmetrization).
+      unsigned int failedOp = 0;
+      MolOps::sanitizeMol(*res, failedOp, MolOps::SANITIZE_CLEANUP);
+      MolOps::detectBondStereochemistry(*res);
+      MolOps::removeHs(*res, false, false);
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/MolFileParser.cpp:3270]: You know my love of explicitness with params

```
bool implicitOnly = true;
bool updateExplicitCount = true;
MolOps::removeHs(*res, implicitOnly, updateExplicitCount);
```

File: Code/GraphMol/FileParsers/MolFileParser.cpp [hunk 3]
```diff
@@ -3446,60 +3432,53 @@ RWMol *MolDataStreamToMol(std::istream *inStream, unsigned int &line,
   Conformer *conf = nullptr;
   try {
     if (ctabVersion == 2000) {
-      fileComplete = FileParserUtils::ParseV2000CTAB(inStream, line, res, conf,
-                                                     chiralityPossible, nAtoms,
-                                                     nBonds, strictParsing);
+      fileComplete = FileParserUtils::ParseV2000CTAB(
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/MolFileParser.cpp:3435]: Perhaps also make the molecule pass by reference.  Understandable if this is a private function though.

File: Code/GraphMol/FileParsers/Mol2FileParser.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/FileParsers/Mol2FileParser.cpp:929]: Sure, I would do that if writing things from scratch, but going back and updating stuff that's not in the public API doesn't feel like a super productive use of time.

File: Code/GraphMol/FileParsers/MolFileParser.cpp (hunk 3 above)
Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolFileParser.cpp:3435]: Agreed that could be fixed at some point. Is that necessary as part of the PR to fix the main API?

Comment (bp-kelley) [reply Code/GraphMol/FileParsers/MolFileParser.cpp:3435]: No, these are private so we can refactor later.

File: Code/GraphMol/FileParsers/Mol2FileParser.cpp (hunk 1 above)
Comment (bp-kelley) [reply Code/GraphMol/FileParsers/Mol2FileParser.cpp:929]: Agreed.

**Comment by greglandrum** [comment]
@bp-kelley did you intended to approve this?

**Review by bp-kelley — APPROVED** [review]

**Pull request merged**